src/Simulation/Simulators/FindNuspecReferences.ps1

```diff
@@ -31,42 +31,73 @@ if (Test-Path $target) {
 $nuspec = [xml](Get-Content "Microsoft.Quantum.Simulators.nuspec.template")
 $dep = $nuspec.CreateElement('dependencies', $nuspec.package.metadata.NamespaceURI)
 
+function Add-PackageReference-IfNew($ref)
+{
+    # Identify package's id either from "Include" or "Update" attribute:
+    $id = $ref.Include
+    $version = $ref.Version
+    
+    if ($id -eq $null -or $id -eq "") {
+        $id = $ref.Update
+    }
+    if ($id.EndsWith('.csproj') -or $id.EndsWith('.fsproj')) 
+    {
+        $id = [System.IO.Path]::GetFileNameWithoutExtension($id)
+    }
+
+    if ($version -eq $null -or $version -eq "") {
+        $version = '$version$'
+    }
+
+    # Check if package already added as dependency, only add if new:
+    $added = $dep.dependency | Where { $_.id -eq $id }
+    if (!$added) {
+        Write-Host "Adding $id (version: $version)"
+        $onedependency = $dep.AppendChild($nuspec.CreateElement('dependency', $nuspec.package.metadata.NamespaceURI))
+        $onedependency.SetAttribute('id', $id)
+        $onedependency.SetAttribute('version', $version)
+    }
+}
 
 # Recursively find PackageReferences on all ProjectReferences:
 function Add-NuGetDependencyFromCsprojToNuspec($PathToCsproj)
 {
+    Write-Host "`nFinding dependencies for $PathToCsproj"
     $csproj = [xml](Get-Content $PathToCsproj)
 
     # Find all PackageReferences nodes:
     $packageDependency = $csproj.Project.ItemGroup.PackageReference | Where-Object { $null -ne $_ }
     $packageDependency | ForEach-Object {
-        # Identify package's id either from "Include" or "Update" attribute:
         $id = $_.Include
-        if ($id -eq $null -or $id -eq "") {
-            $id = $_.Update
-        }
-
-        # Check if package already added as dependency, only add if new:
-        $added = $dep.dependency | Where { $_.id -eq $id }
-        if (!$added) {
-            Write-Host "Adding $id"
-            $onedependency = $dep.AppendChild($nuspec.CreateElement('dependency', $nuspec.package.metadata.NamespaceURI))
-            $onedependency.SetAttribute('id', $id)
-            $onedependency.SetAttribute('version', $_.Version)
-        }
+        Write-Host "Detected package dependencies: $id"
+    }
+
+    $packageDependency | ForEach-Object {
+        Add-PackageReference-IfNew $_ $dep
     }
 
-    # Recursively check on project references:
     $projectDependency = $csproj.Project.ItemGroup.ProjectReference | Where-Object { $null -ne $_ }
     $projectDependency | ForEach-Object {
+        $id = $_.Include
+        Write-Host "Detected project dependencies: $id"
+    }
+
+    # Assume there is a package for project references that are not tagged as to be included in the simulator package:
+    $projectDependency | Where-Object {$_.IncludeInSimulatorPackage -ne 'true' -and $_.IsQscReference -ne 'true'} | ForEach-Object {
+        Add-PackageReference-IfNew $_ $dep
```

Comment:
There's two parameters passed here, but `Add-PackageReference-IfNew` only declares one formal parameter?

Reply:
I pretty much copied the signature of the method below - either one is fine with me.

Reply:
removed it

Reply:
Ah, I thought it was new from the way it showed up in diffs. My apologies!